app/src/main/java/com/yeonae/chamelezone/CheckDialogFragment.kt

```diff
@@ -81,8 +76,38 @@ class CheckDialogFragment : DialogFragment() {
         return inflater.inflate(R.layout.fragment_check_dialog, container, false)
     }
 
+    private fun addKeywordView(keyword: String) {
+        val layout = LayoutInflater.from(context)
+            .inflate(R.layout.item_check_box, check_box, false)
+        check_box.addView(layout)
+        layout.findViewById<CheckBox>(R.id.cb_keyword).text = keyword
+        selectedKeyword.forEach {
+            if (it == keyword) {
+                layout.findViewById<CheckBox>(R.id.cb_keyword).isChecked = true
+            }
+        }
+        layout.findViewById<CheckBox>(R.id.cb_keyword).setOnClickListener {
+            if (layout.findViewById<CheckBox>(R.id.cb_keyword).isChecked) {
+                selectedKeyword.add(layout.findViewById<CheckBox>(R.id.cb_keyword).text.toString())
+            } else if (!layout.findViewById<CheckBox>(R.id.cb_keyword).isChecked) {
+                selectedKeyword.remove(layout.findViewById<CheckBox>(R.id.cb_keyword).text.toString())
+            }
+        }
```

Comment:
```suggestion
        layout.findViewById<CheckBox>(R.id.cb_keyword).run{ checkBox->
        setOnClickListener {
            if (checkBox.isChecked) {
                selectedKeyword.add(checkBox.text.toString())
            } else if (!checkBox.isChecked) {
                selectedKeyword.remove(checkBox.text.toString())
            }
        }
        }
```
findViewById는 무거운 행동입니다. 위에 포함해서 한번만 하도록 개선할 수 있을 것 같아요.

Reply:
수정했습니다